app/pdf_utils/fill_sections.py

```diff
@@ -32,51 +42,78 @@ def reference(reference_id: str) -> NoEscape:
 
 
 def get_paragraph_text(section_element):
     """
     Utility function used in generator.py to get text input by user
     """
-    for _indx, paragraph in enumerate(section_element["children"]):
+    for _indx, text_leaf in enumerate(section_element["children"]):
+
+        # check if the text_leaf has the path text
+        section_p_text = pydash.get(obj=text_leaf, path="text")
 
-        # check if the paragraph has the path text
-        # if so get text
-        section_p_text = pydash.get(obj=paragraph, path="text")
+        if len(section_element["children"]) > 1:
 
-        # TODO get line breaks i.e. type='p', children.0.text = ''
+            # use child as a "counter"
+            for child in section_element["children"]:
 
-        # check if the paragraph has type 'ref'
-        paragraph_type = pydash.get(obj=paragraph, path="type")
+                # if we have type 'ref', get the reference id and process reference
+                child_type = pydash.get(obj=child, path="type")
 
-        # if we have type 'ref', get the reference id and process reference
-        if paragraph_type == "ref":
+                # get reference text from text_leaf
+                ref_text = pydash.get(obj=text_leaf, path="text")
 
-            refId = pydash.get(obj=paragraph, path="refId")
+                # if we have type 'ref', get the reference id and process reference(s), checking for multiple references
+                if child_type == "ref":
 
-            # we will return the reference content if encountered
-            return reference(refId)
+                    ref_ids: list = get_reference_ids(section_element["children"])
 
-        # TODO allow inline text formatting
-        # # isolate full text leaf element
-        # text_leaf = paragraph
-        # # use functions in TEXT_WRAPPERS to format text
-        # for option, command in TEXT_WRAPPERS.items():
+                    # join the formatted references as a single string
+                    ids_ = " ".join(str(i) for i in ref_ids)
 
-        #     if option in text_leaf.keys():
-        #         # if text_leaf.get(option) and e.strip(" ") != "":
-        #         section_p_text = section_p_text.strip(" ") if section_p_text.strip(" ") != "" else section_p_text
+                    # return reference text and bibliography refs
+                    return NoEscape(f"{ref_text} {ids_}")
 
-        #         # escape latex
-        #         section_p_text = utils.escape_latex(section_p_text)
+        # use functions in TEXT_WRAPPERS to format text
+        for option, _command in TEXT_WRAPPERS.items():
 
-        #         # run functions in text_wrappers
-        #         section_p_text = command(section_p_text)
+            # if the option in text_leaf is true
+            if text_leaf.get(option):
+                # run functions in text_wrappers
+                section_p_text = wrap_text(text_leaf)
 
-    return section_p_text
+        return section_p_text
```

Comment:
Hey @naomatheus, this section seems to be complex. Maybe we can simplify this?
We are iterating section_element["children"] twice.

Trying to simplify we can get something like this. Is this expected?
```python
    ref_ids: list = get_reference_ids(section_element["children"])

    for text_leaf in section_element["children"]:
        # check if the text_leaf has the path text
        section_p_text = pydash.get(obj=text_leaf, path="text")

        if ref_ids:  # If there are ref ids
            ids_ = " ".join(str(i) for i in ref_ids)
            # return reference text and bibliography refs
            return NoEscape(f"{section_p_text} {ids_}")

        # XXX: Only this is required right for wrapping text?
        section_p_text = wrap_text(text_leaf)

        # XXX: Asumming there are only 1 children?
        return section_p_text
```